collector/project.go

```diff
@@ -63,68 +65,102 @@ func (p Project) Collect(c *CollectorOpts) interface{} {
+				p.Ns.Update(totalNs)
+				nsUtils = append(nsUtils, float64(totalNs))
+				p.Ns.UpdateDetails(nsCollection.Data)
+				log.Debugf("    Found %d namespaces", totalNs)
 			}
-			nsCollection := filterNSCollectionWithProjectID(GetNamespaceCollection(c, cluster.Links["namespaces"]), projectID)
-			totalNs := len(nsCollection.Data)
-			p.Ns.Update(totalNs)
-			nsUtils = append(nsUtils, float64(totalNs))
-			p.Ns.UpdateDetails(&nsCollection)
+		}
+
+		projectClient, err := GetProjectClient(c, project.ID)
+		if err != nil {
+			log.Errorf("Failed to get project client ID %s err=%s", project.ID, err)
+			continue
 		}
 
 		// Workload
-		wlCollection := GetWorkloadCollection(c, project.Links["workloads"])
-		if wlCollection != nil {
+		log.Debugf("  Collecting Workloads")
+		wlCollection, err := projectClient.Workload.ListAll(&nonRemoved)
+		if err != nil {
+			log.Errorf("Failed to get Workload for project %s err=%s", project.ID, err)
+		} else {
 			totalWl := len(wlCollection.Data)
 			p.Workload.Update(totalWl)
 			wlUtils = append(wlUtils, float64(totalWl))
+			log.Debugf("    Found %d Workloads", totalWl)
 		}
 
 		// Pipeline
-		pipelineCollection := GetPipelineCollection(c, project.Links["pipelines"])
-		if pipelineCollection != nil {
+		log.Debugf("  Collecting Pipelines")
+		pipelineCollection, err := projectClient.Pipeline.ListAll(&nonRemoved)
+		if err != nil {
+			log.Errorf("Failed to get Pipelines for project %s err=%s", project.ID, err)
+		} else {
 			p.Pipeline.TotalPipelines += len(pipelineCollection.Data)
+			log.Debugf("    Found %d Pipelines", p.Pipeline.TotalPipelines)
 		}
 
 		// Source provider
-		sourceCollection := GetSourceCodeProviderCollection(c, project.Links["sourceCodeProviders"])
-		if sourceCollection != nil {
+		log.Debugf("  Collecting SourceCodeProviders")
+		sourceCollection, err := projectClient.SourceCodeProvider.ListAll(&nonRemoved)
+		if err != nil {
+			log.Errorf("Failed to get SourceCodeProvider for project %s err=%s", project.ID, err)
+		} else {
 			p.Pipeline.Enabled = 1
 			for _, provider := range sourceCollection.Data {
 				p.Pipeline.SourceProvider.Increment(provider.Type)
 			}
+			log.Debugf("    Found %d SourceCodeProviders", len(sourceCollection.Data))
 		}
 
 		// HPA
-		hpaCollection := GetHPACollection(c, project.Links["horizontalPodAutoscalers"])
-		if hpaCollection != nil {
+		log.Debugf("  Collecting HPAs")
+		hpaCollection, err := projectClient.HorizontalPodAutoscaler.ListAll(&nonRemoved)
+		if err != nil {
+			log.Errorf("Failed to get HPA for project %s err=%s", project.ID, err)
+		} else {
 			totalHPAs := len(hpaCollection.Data)
 			p.HPA.Update(totalHPAs)
 			hpaUtils = append(hpaUtils, float64(totalHPAs))
+			log.Debugf("    Found %d HPAs", totalHPAs)
 		}
 
 		// Pod
-		poCollection := GetPodCollection(c, project.Links["pods"])
-		if poCollection != nil {
+		log.Debugf("  Collecting Pods")
+		poCollection, err := projectClient.Pod.ListAll(&nonRemoved)
```

Comment:
nit: `pod` not `po`.

Reply:
I didn't change varibles name, i let it like was.